operator/validator/event_handler.go

```diff
@@ -379,12 +383,16 @@ func (c *controller) handleClusterReactivatedEvent(
 	}
 
 	if ongoingSync && len(toEnable) > 0 {
+		var started bool
 		for _, share := range toEnable {
-			_, err = c.onShareStart(logger, share)
+			started, err = c.onShareStart(logger, share)
 			if err != nil {
 				logger.Warn("could not start validator", zap.String("pubkey", hex.EncodeToString(share.ValidatorPubKey)), zap.Error(err))
 			}
 		}
+		if started {
+			c.indicesChange <- true
+		}
```

Comment:
what if last one in range wasn't started, but 2 befores did. we still need to shoot an event to the channel. this won't work. last one is what matters in this code. I suggest maybe stop changing started after one has returned true.

Reply:
good catch